File: app/src/main/java/com/projects/johnny/myway/RequestLocationActivity.java
package com.projects.johnny.myway;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

/**
 * This class simply exists to tell users to turn on their locations.
 */
public class RequestLocationActivity extends AppCompatActivity {

    private Button SignOutButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_request_location);

        SignOutButton = (Button) findViewById(R.id.sign_out_button);
        SignOutButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                intent.setFlags(intent.getFlags() | Intent.FLAG_ACTIVITY_NO_HISTORY);
                startActivity(intent);
            }
        });
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
        intent.setFlags(intent.getFlags() | Intent.FLAG_ACTIVITY_NO_HISTORY);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/projects/johnny/myway/RequestLocationActivity.kt
package com.projects.johnny.myway

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_request_location.*

/*
    This class simply exists to tell users to turn on their locations setting.
 */
class RequestLocationActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_request_location)

        signOutButton.setOnClickListener {
            val intent = Intent(applicationContext, MainActivity::class.java)
            intent.setFlags(intent.flags or Intent.FLAG_ACTIVITY_NO_HISTORY)
            startActivity(intent)
        }


    }

    override fun onBackPressed() {
        super.onBackPressed()

        val intent = Intent(applicationContext, MainActivity::class.java)
        intent.setFlags(intent.flags or Intent.FLAG_ACTIVITY_NO_HISTORY)
        startActivity(intent)
        finish()
    }
}